sdk/storage/azure-storage-files-datalake/CMakeLists.txt

```
@@ -4,7 +4,7 @@
 cmake_minimum_required (VERSION 3.13)
 
 set (AZURE_STORAGE_DATALAKE_HEADER
-    inc/azure/storage/files/datalake/datalake.hpp
+    inc/azure/storage/datalake.hpp
```

do you think this should be `file_datalake.hpp` or just `datalake.hpp`?
also `file_shares.hpp` or `shares.hpp`?

---

Keep the `/files/` subdirectory.
```suggestion
    inc/azure/storage/files/datalake.hpp
```

---

While, I watched the replay again, this is not we agreed on in the meeting this week.

---

what we've agreed on is `inc/azure/storage/datalake.hpp`

---

I see what you are saying. Given the "package name" already contains `azure-storage-files-datalake`, having it show up again in the include-path _seems_ redundant.

However, moving it all the way to the root, like you observed, introduces the question of whether we need to keep `files` in the header name or not.

I don't feel too strongly about this, but my mental model when bringing this up was to limit larger directory structure changes and primarily focus on simplifying the include header uses case from the user.

`#include <azure/storage/files_datalake.hpp>` vs `#include <azure/storage/files/datalake.hpp>` isn't materially different to warrant moving datalake.hpp up anymore than just one level, and there is value in keeping the `/files/` structure clear from an organizational perspective. There's another small benefit by doing this, and that's avoiding underscore in the simplified header.

I think where we have landed in this PR now is great :)